presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/DesugaringOptimizer.java

```diff
@@ -172,6 +182,14 @@ public PlanNode visitValues(ValuesNode node, RewriteContext<Void> context)
                             .collect(toImmutableList()));
         }
 
+        @Override
+        public PlanNode visitApply(ApplyNode node, RewriteContext<Void> context)
+        {
+            PlanNode input = context.rewrite(node.getInput());
+            PlanNode subquery = context.rewrite(node.getSubquery());
+            return new ApplyNode(node.getId(), input, subquery, node.getSubqueryAssignments(), node.getCorrelation());
```

Comment:
Please add comment here (something similar to what you had in the commit message). Otherwise, it likely would lead people to believe that missing the rewrite is an oversight.